Sources/Widgets/Widgets3D/ResliceCursorWidget/behavior.js

```diff
@@ -83,36 +85,93 @@ export default function widgetBehavior(publicAPI, model) {
+      currentCenter[2] + direction * normal[2] * imageSpacing[2],
     ];
-
-    if (!boundingBox.containsPoint(...newCenter)) {
-      // Clamp the center to the bounds of the image
-      const imageOrigin = image.getOrigin();
-      const localRoundedCenter = currentCenter.map(
-        (value, index) =>
-          Math.round(value - imageOrigin[index]) / imageSpacing[index]
-      );
-
-      // Get it back in world coordinates
-      newCenter = localRoundedCenter.map(
-        (value, index) => value * imageSpacing[index] + imageOrigin[index]
-      );
+    const newCenterInsideBounds = bounds.containsPoint(...newCenter);
+
+    // The center is out of the image
+    if (!newCenterInsideBounds) {
+      // Find which plane is out of the bounds
+      const isCurrentPlaneOutOfBounds =
+        !vtkBox.intersectPlane(imageBounds, newCenter, normal) &&
+        vtkBox.intersectPlane(imageBounds, currentCenter, normal);
+
+      const otherViewsPlaneOutOfBounds = [];
+      let otherPlanesAreOutsideBounds = false;
+      for (let i = 0; i < otherViews.length; i++) {
+        const outBounds =
+          !vtkBox.intersectPlane(imageBounds, newCenter, otherNormals[i]) &&
+          vtkBox.intersectPlane(imageBounds, currentCenter, otherNormals[i]);
+        otherViewsPlaneOutOfBounds.push(outBounds);
+        otherPlanesAreOutsideBounds = otherPlanesAreOutsideBounds && outBounds;
+      }
+
+      // Basic clamping of the newCenter to the image bounds
+      if (!otherPlanesAreOutsideBounds && isCurrentPlaneOutOfBounds) {
+        // Find new reslice cursor center in local coordinates and round it to be on a valid slice
+        const imageOrigin = image.getOrigin();
+        const localRoundedCenter = currentCenter.map(
+          (value, index) =>
+            Math.round(value - imageOrigin[index]) / imageSpacing[index]
+        );
+
+        // Get it back in world coordinates
+        newCenter = localRoundedCenter.map(
+          (value, index) => value * imageSpacing[index] + imageOrigin[index]
+        );
+      } else {
+        // Need to find which axis will be out of the bounds
+        for (let i = 0; i < otherViews.length; i++) {
+          const isOut = otherViewsPlaneOutOfBounds[i];
+          if (isOut) {
+            const outViewName = otherViews[i];
+            const displayViewOutOfBounds =
+              i === 0 ? otherViews[1] : otherViews[0];
+            // Get the correct line that will be used to find the new center
+            // by computing the intersection between current plane and the line
+            const axisName = `Axis${outViewName}in${displayViewOutOfBounds}`;
+            const axis = model.widgetState[`get${axisName}`]();
+            newCenter = vtkPlane.intersectWithLine(
+              axis.getPoint1(),
+              axis.getPoint2(),
+              newCenter,
+              normal
+            ).x;
+            break;
+          }
+        }
+      }
```

Comment:
couldn't you use `vtkBox.intersectBox` ? (maybe its naming is not good and should be changed (by preserving backward compatibility)).

Reply:
I don't want to intersect with a box. I really want to get the intersection between a plane and a line.
Why would you use `vtkBox.intersectBox` ?

Reply:
If newCenter is outside the box, you "project" `newCenter` onto the box (the box being the bounds).
The projection is done by creating a line betwen `oldCenter` and `newCenter` and see where it intersect the bounds. `newCenter` is then moved onto the intersection point.

Reply:
Looks like ResliceCursorWidget/helpers.js `boundPoint` does what you want, no ?

Reply:
Maybe but I'm not sure. As it works like this, I don't want to change because I'm afraid it would take me too many times to think on an other way to use this method.